### ISSUE TITLE
Explore form validation in Angular

### ISSUE THREAD

**Issue by xpdavid** [TEAMMATES/teammates#9419]
When migrating, most of the form validation logics are discarded (due to time constraint). The data is still being validated but only in backend. The error message is not user friendly (see #9417).

We need to explore form validation options https://angular.io/guide/form-validation in Angular to provide more user-friendly experience.

Related issue: https://github.com/TEAMMATES/teammates/issues/8931 https://github.com/TEAMMATES/teammates/issues/8597

**Comment by SahibArora** [comment]
Hi, 
I have some experience with Angular, I might be able to help here, if its still open?
Thank you

**Comment by monmanuela** [comment]
Hi @SahibArora, thank you for your interest to contribute to TEAMMATES ☺️ I suggest you to look at `d.FirstTimers` issues first to familiarize yourself with the codebase and the contributing workflow. You can come back to this issue later when you are more familiarized with the codebase.

**Comment by madanalogy** [comment]
I figured it might be useful to compile a list of forms that could use front-end validation. I only explored pages in a MSS use case exploration of the application. If anyone has additional forms/pages to recommend let me know and I'll add it to the list.

- [ ] Add Course Form (Might be addressed by #9820) 
- [ ] Enrol Students Page (Might be better just to have static display of requirements)
- [ ] Edit Course Details Page
- [X] Edit Student Details Page (Completed in #9633)

I didn't add any of the session/question forms as they already have extensive guidance on their varying functions/requirements, and so the impact of front-end validation might not be as strong as those outlined above.
